i6_native_ops/fbw2/fbw2_op.cu

```diff
@@ -769,8 +784,8 @@ std::vector<torch::Tensor> fbw2_cuda(torch::Tensor& num_states, torch::Tensor& n
     else {
         start_dev_kernel2(baum_welch_v2<true>, n_blocks_fbw, block_size_fbw, 0,
                           (n_frames, n_seqs, n_emissions, n_edges,
-                           state_offsets.data().get(), edge_offsets.data().get(), d_seq_lens,
-                           d_from, d_to, d_weights, d_emission_idxs, d_start_states, d_end_states,
+                           state_offsets.data().get(), d_edge_offsets.data().get(), d_seq_lens,
+                           d_from, d_to, d_weights, d_emission_idxs, d_start_states, d_end_states, d_end_state_offsets,
```

Comment:
since the new parameter does not work for the v1 `baum_welch` implementation, should we add some check somewhere that `debug_options.explicit_merge` is not set when we have multiple ends?

Reply:
Yes that's a good idea. Alternatively I could adjust the `v2` accordingly.